### PR TITLE
Bhr 53

### PR DESCRIPTION
**Pull request by RobertHebel**
redux action and reducer

### CODE REVIEW

File: frontend/test/reducers/borrowReducer.test.js [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+        error: 'This is an example error message',
+      };
+      const newState = borrowReducer(state, action);
+      expect(newState).to.be.deep.equal({
+        ...state,
+        isLoading: {
+          gettingData: false,
+        },
+        errors: {
+          gettingData: 'This is an example error message',
```
Comment (szymonbrainhub) [inline frontend/test/reducers/borrowReducer.test.js]: It's usually a good idea to keep such texts in a variable. That way You are sure, that there is no typo.

File: frontend/test/reducers/borrowReducer.test.js [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+        error: 'This is an example error message',
+      };
+      const newState = borrowReducer(state, action);
+      expect(newState).to.be.deep.equal({
+        ...state,
+        isLoading: {
+          reservingBook: false,
+        },
+        errors: {
+          reservingBook: 'This is an example error message',
```
Comment (szymonbrainhub) [inline frontend/test/reducers/borrowReducer.test.js]: Same as above, typo prevent 😀.

File: frontend/test/reducers/borrowReducer.test.js [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+        error: 'This is an example error message',
+      };
+      const newState = borrowReducer(state, action);
+      expect(newState).to.be.deep.equal({
+        ...state,
+        isLoading: {
+          updatingBook: false,
+        },
+        errors: {
+          updatingBook: 'This is an example error message',
```
Comment (szymonbrainhub) [inline frontend/test/reducers/borrowReducer.test.js]: Same as above

File: frontend/test/reducers/borrowReducer.test.js [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+        error: 'This is an example error message',
+      };
+      const newState = borrowReducer(state, action);
+      expect(newState).to.be.deep.equal({
+        ...state,
+        isLoading: {
+          cancelingRequest: false,
+        },
+        errors: {
+          cancelingRequest: 'This is an example error message',
```
Comment (szymonbrainhub) [inline frontend/test/reducers/borrowReducer.test.js]: Same as above

File: frontend/src/reducers/borrowReducer.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+        available: action.result.body.available,
+        queueLength: action.result.body.queue,
+        positionInQueue: action.result.body.position,
+        isBookBorrowed: action.result.body.isBookBorrowed,
+        isLoading: {
+          gettingData: false,
+        },
+        errors: {
+          gettingData: null,
+        },
```
Comment (RobertHebel) [inline frontend/src/reducers/borrowReducer.js]: it looks nice, but it also violates the eslint rule 'no-case-declarations'

File: frontend/src/reducers/borrowReducer.js [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+  UPDATE_BOOK_STATUS_REQUEST,
+  UPDATE_BOOK_STATUS_SUCCESS,
+  UPDATE_BOOK_STATUS_ERROR,
+  CANCEL_BOOK_RESERVATION,
+  CANCEL_BOOK_RESERVATION_SUCCESS,
+  CANCEL_BOOK_RESERVATION_ERROR,
+} from '../constants/actionTypes';
+import initialState from './initialState';
+
+export default function(state = initialState.borrow, action) {
```
Comment (RobertHebel) [inline frontend/src/reducers/borrowReducer.js]: fixed

File: frontend/src/reducers/borrowReducer.js [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+          cancelingRequest: false,
+        },
+        errors: {
+          cancelingRequest: action.error,
+        },
+      };
+    default:
+      return state;
+  }
+}
```
Comment (RobertHebel) [inline frontend/src/reducers/borrowReducer.js]: fixed

File: frontend/test/reducers/borrowReducer.test.js (hunk 1 above)
Comment (RobertHebel) [inline frontend/test/reducers/borrowReducer.test.js]: fixed

File: frontend/test/reducers/borrowReducer.test.js (hunk 2 above)
Comment (RobertHebel) [inline frontend/test/reducers/borrowReducer.test.js]: fixed

File: frontend/test/reducers/borrowReducer.test.js (hunk 3 above)
Comment (RobertHebel) [inline frontend/test/reducers/borrowReducer.test.js]: fixed

File: frontend/test/reducers/borrowReducer.test.js (hunk 4 above)
Comment (RobertHebel) [inline frontend/test/reducers/borrowReducer.test.js]: fixed

**Pull request merged**